git-filter-repo

```diff
@@ -3357,6 +3357,11 @@ class RepoFilter(object):
     commit.file_changes = [v for k,v in sorted(new_file_changes.items())]
 
   def _tweak_commit(self, commit, aux_info):
+    # Callbacks are provided environment variables related to commit
+    # similarly to filter-branch
+    # https://git-scm.com/docs/git-filter-branch#_filters
+    os.environ["GIT_COMMIT"] = commit.original_id.decode('utf-8')
```

Comment:
Environment variables are a way of passing information between parent and child processes. Callbacks run in the same process, so using environment variables to pass information around seems like a wasteful way to pass the information.  Especially since a commit callback can just access commit.original_id directly.